Sources/DateTime/InternalUTCClock.swift

```diff
@@ -31,33 +31,68 @@ public struct InternalUTCClock {
         }
 
         public init(secondsComponent: Int64, attosecondsComponent: Int64) {
-            self._value = Swift.Duration(secondsComponent: secondsComponent, attosecondsComponent: attosecondsComponent)
+            _value = Swift.Duration(secondsComponent: secondsComponent, attosecondsComponent: attosecondsComponent)
         }
 
         @available(*, deprecated, renamed: "seconds")
         public func secondsSinceEpoch() -> Int64 {
-            return self._value.components.seconds
+            _value.components.seconds
         }
 
         public func seconds() -> Int64 {
-            return self._value.components.seconds
+            _value.components.seconds
         }
-        
+
         public func attoseconds() -> Int64 {
-            return self._value.components.attoseconds
+            _value.components.attoseconds
         }
     }
 
     public init() {}
 }
 
+private extension String {
+    func pad(_ padding: Int = 2) -> String {
+        let toPad = padding - count
+        if toPad < 1 {
+            return self
+        }
+        return "".padding(toLength: toPad, withPad: "0", startingAt: 0) + self
+    }
+
+    func padMicroseconds(_ padding: Int = 6) -> String {
```

Comment:
```suggestion
    func padMicroseconds(_ padding: Int = 6) -> String { pad(padding) }
```
?

Reply:
Nope, for micros we want to pad at the end - it's the opposite from YMDHMS padding...

Reply:
Hmm, I added the following to `testOutput`:
```
diff --git a/Tests/DateTimeTests/DateTimeTests.swift b/Tests/DateTimeTests/DateTimeTests.swift
index fa206c8..c881866 100644
--- a/Tests/DateTimeTests/DateTimeTests.swift
+++ b/Tests/DateTimeTests/DateTimeTests.swift
@@ -66,5 +66,13 @@ final class DateTimeTests: XCTestCase {
         XCTAssert(fractionsOfSeconds == "123456")
 
         print("\(instant)")
+
+        instant = InternalUTCClock.Instant(secondsComponent: now.seconds(),
+                                           attosecondsComponent: 000_456_000_000_000_000)
+        description = "\(instant)"
+        fractionsOfSeconds = description.dropFirst("2022-03-10 14:15:56.".count)
+        XCTAssert(fractionsOfSeconds == "000456")
+
+        print("\(instant)")
     }
 }
```

and it fails

```
/home/sbn/GitHub/package-datetime/Tests/DateTimeTests/DateTimeTests.swift:74: error: DateTimeTests.testOutput : XCTAssertTrue failed -
2023-02-10 13:40:09.456000
```

Reply:
Argh, you are correct, many thanks. Updated PR:

https://github.com/ordo-one/package-datetime/pull/9/